### PR TITLE
CoreData: fix missing comma

### LINKED ISSUE #675: CoreData: error in swift 4 template
Line 145 of core data's swift4.stencil should end with a comma (since it's an array)

### PR DESCRIPTION
**Pull request by djbe**
Fixes #675.

Note: I intentionally did not apply this change to the swift 3 templates, as we're dropping them in #691. CI will probably fail because of this.

### CODE REVIEW

File: templates/coredata/swift4.stencil [hunk 1]
```diff
@@ -179,7 +179,7 @@ extension {{ entityClassName }} {
     let model = persistentStoreCoordinator.managedObjectModel
     let substitutionVariables: [String: Any] = [
       {% for variableName, variableType in fetchRequest.substitutionVariables %}
-      "{{ variableName }}": {{ variableName | lowerFirstWord }}
+      "{{ variableName }}": {{ variableName | lowerFirstWord }}{{ ", " if not forloop.last }}
```
Comment (AliSoftware) [inline templates/coredata/swift4.stencil]: TIL that Stencil supported the `{{ x if y }}` ruby-like construct, way more compact than the `{% if not forloop.last %}, {% endif %}`, I missed that, 👍 

We should update all other templates to use that construct instead – for those not already using it that is, i.e. plist, json, yaml, and the strings-objc ones

**Review by AliSoftware — APPROVED** [review]
LGTM!

We should merge #691 first, so that the CI will end up green on this one and make it mergeable (which is why you made it a Draft I guess?)

A bit out of scope but we could also use the occasion of that PR to convert every remaining `{% if not forloop.last %}, {% endif %}` in other templates too with the more compact `{{ ", " if not forloop.last }}`

**Comment by SwiftGen-Eve** [comment]
Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: templates/coredata/swift4.stencil (hunk 1 above)
Comment (AliSoftware) [reply templates/coredata/swift4.stencil]: Fixed in https://github.com/SwiftGen/SwiftGen/pull/693

**Pull request merged**